tests/subsys/net/lib/fota_download/src/test_fota_download.c

```diff
@@ -240,76 +229,87 @@ void client_callback(const struct fota_download_evt *evt)
 	}
 }
 
+
+/**
+ * @brief Reset mocks and initialize the fota_download library
+ *
+ * Call this before every test
+ */
 static void init(void)
 {
 	int err;
 
-	k_sem_init(&download_with_offset_sem, 0, 1);
 	start_with_offset = false;
 	fail_on_offset_get = false;
 	fail_on_connect = false;
 	fail_on_start = false;
+	download_client_start_file = NULL;
+	spm_s0_active_retval = false;
 
 	err = fota_download_init(client_callback);
 	zassert_equal(err, 0, NULL);
 }
 
-static void test_fota_download_start(void)
+
+/**
+ * @brief Generic test of fota_download
+ * Not to be called directly by the test suite. Performs a parameterized test of fota_download_start
+ *
+ * @param resource_locator - The resource locator to be fed to fota_download_start
+ * @param expected_selection - The resource locator that fota_download_start is expected to select
+ * @param s0_active - If true, pretend that s0 is active. Otherwise, pretend that s1 is active.
+ */
+static void test_fota_download_start_generic(const char * const resource_locator,
```

Comment:
Git absolutely butchers the diff on this function, ignore diff and treat like a brand new function

Reply:
Yes, it's very difficult to follow.